modules/core/04-channel/keeper/upgrade.go

```diff
@@ -597,6 +597,94 @@ func (k Keeper) startFlushUpgradeHandshake(
 	return nil
 }
 
+// ChanUpgradeOpen is called by a module to complete the channel upgrade handshake and move the channel back to an OPEN state.
+// This method should only be called after both channels have flushed any in-flight packets.
+func (k Keeper) ChanUpgradeOpen(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	counterpartyChannelState types.State,
+	proofChannel []byte,
+	proofHeight clienttypes.Height,
+) error {
+	channel, found := k.GetChannel(ctx, portID, channelID)
+	if !found {
+		return errorsmod.Wrapf(types.ErrChannelNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+
+	if k.hasInflightPackets(ctx, portID, channelID) {
+		return errorsmod.Wrapf(types.ErrPendingInflightPackets, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
```

Comment:
this check could be at the top of the function which would be a micro optimisation on gas I guess. I think the spec has it at top also, so unsure if that's what was intended